draft-frindell-moq-chat.md

```diff
@@ -55,115 +55,81 @@ receive messages published by others.
 Commonly used terms in this document are described below.
 
 
-# Catalog Format
-
-The moq-chat catalog is used to determine the list of participants in the chat.
-The format is a text file with text delta encodings.
-
-Object 0 in any Group is a complete list of the usernames of the participants in
-the chat at the time the Object was published, one per-line.  The first line of
-Object 0 is the version of the catalog format specificaiton. The format
-described here is moq-chat catalog format version 1.
-
-Object 0 Example:
-
-~~~
-version=1
-alice
-bob
-charlie
-~~~
-
-Any Object in a Group with sequence number higher than 0 is a delta encoding,
-with new participants prefaced with a '+' character and participants that have
-left the chat prefaced with a '-' character.  Every delta encoding object MUST
-be applied for an endpoint to have a current list of participants.
-
-Delta Encoding Exmaple:
-
-~~~
-+daphne
--bob
-~~~
-
-Every username in the catalog MUST be unique.  If an endpoint receives a catalog
-or delta encoding that would result in the same username more than once, it MUST
-close the session with an error.
-
-Every object in a catalog group MUST be on a single stream, so that updates are
-delivered in order.
-
 # Chat Operation
 
-## MoQ Chat Server
+## MoQ Relay
 
-The protocol requires that one entity operate as a chat server.  The function
-of the server is to maintain the list of participants in the chat and publish
-catalog updates.
+The protocol requires a MoQ relay to act as a chat server.  The relay maintains
+the set of connected clients that have announced a chat track.
 
 ## Chat ID
 
 Every chat has a unique ID.  The ID is a string of arbitrary length and uniquely
-identifies the chat.  The creation of chat IDs and discovery of the server
-endpoint is out of the scope of this document.
+identifies the chat.  The creation of chat IDs and discovery of the relay is
+out of the scope of this document.
 
 ## Track Names
 
-The catalog for a given chat is available by subscribing to moq-chat/id.
+Each chat participant has a chat track.  The namespace of the track is
+("moq-chat", \<id\>, \<user-id\>, \<device-id\>, \<timestamp\>) and the name
+track is "chat".
 
-## Joining the Chat
+* id - the ID of the chatroom
+* user-id - the user ID
+* device-id - a unique identifier for each device for the user.  This allows
+              the same user to join the chat from multiple devices
+* timestamp - the timestamp in seconds when the track started.  This allows a
```

Comment:
Are there any restrictions besides that it monotonically increase? Can one publisher start at 1970 and the other at 2023? What if one uses seconds and another milliseconds.

I  imagine that if the units are too big you'll have collisions, but aside from that I think publishers can do whatever they want?

Reply:
I didn't put any restrictions here yet since there's no purpose for them yet.  I could say it needs to be seconds since 1970 or something, which would facilitate loading chat history without a catalog or timeline track.

Do you want me to make it restrictive?

Reply:
No; I just think the spec should be explicit about it.